lib/supavisor/secret_checker.ex

```diff
@@ -0,0 +1,117 @@
+defmodule Supavisor.SecretChecker do
+  @moduledoc false
+
+  use GenServer
+  require Logger
+
+  alias Supavisor.Helpers
+
+  @interval :timer.seconds(15)
+
+  def start_link(args) do
+    name = {:via, Registry, {Supavisor.Registry.Tenants, {:secret_checker, args.id}}}
```

Comment:
Do we need that registry at all? It doesn't seem to be used for anything.

Reply:
yep, it can be very useful during debugging

Reply:
I am not 100% sure if it is good approach. Maybe we could use [`Process.set_label/1`](https://hexdocs.pm/elixir/Process.html#set_label/1) in the future instead?